roles/sriov-nic-init/files/dpdk-devbind.py

```diff
@@ -0,0 +1,741 @@
+#! /usr/bin/env python
```

Comment:
can we avoid having this here? multiple reasons - this script had some serious compatibility issues in the past with python versions, there are hardcoded kernel modules names, hardcoded pci device models, it often returns misleading error message and it'll be a pain to maintain it here separately

if we need it only to bind VFs to kernel modules, we can write directly to `/sys/bus/pci/drivers/{{driver_name}}/bind` instead (we're already doing stuff directly in the `/sys` fs in the same role already)

in worst case we could just clone dpdk and call it from there

Reply:
Hi @przemeklal Thanks for the review.
This script included here with full source code and licensing headers unchanged. It is used as a simple tool to do the 'job' instead of cloning entire dpdk in each hosts (this is how it's being done at the moment with `roles/dpdk-install`). Internally it also uses the same `drivers/{{driver_name}}/bind` you've suggested. Actually there are few other steps involved to do the manual attachment using driver bind interface and this script takes care of those.

My personal preference would be to use an ansible module with [driverctl](https://gitlab.com/driverctl/driverctl). The driverctl lets overriding the default driver for devices which is then persisted on system reboot. But unfortunately it is not available for all distros unless we build it from source.